amd_openvx_extensions/amd_media/decoder.cpp

```diff
@@ -520,6 +534,7 @@ vx_status CLoomIoMediaDecoder::Initialize()
             return -1;
         if (useVaapi[mediaIndex]) {
             codecContext->get_format  = get_hw_format;
+            int dev_id = (hwDeviceID[mediaIndex] < 0) ? mediaIndex : hwDeviceID[mediaIndex]; 
```

Comment:
where do you use the dev_id? did you mean to pass it to the hw_decoder_init function?

Reply:
Yes. Good catch